constants.cs

```diff
@@ -76,15 +77,31 @@ public static class Constants {
 
     // Error messages
     public const string ERROR_FILE_FORMAT = "ERROR: File format does not conform to format: <filename>.mp; ";
+
     public const string ERROR_DISPATCHER_DIGIT =
         "ERROR: Dispatcher done goofed. Passed {0} to the digit FSA, but {0} is not a digit.";
     public const string ERROR_DISPATCHER_PUNCTUATION =
         "ERROR: Dispatcher done goofed. Passed {0} to the punctuation FSA, but {0} is not punctuation.";
+    public const string ERROR_DISPATCHER_LETTERS =
+        "ERROR: Dispatcher done goofed. Passed {0} to the letters FSA, but {0} is not a letter";
 
     // List of whitespace characters
     public const string WHITESPACE = " \r\n\t";
     // List of digit characters
     public const string DIGITS = "0123456789";
+    
     // List of punctuation characters
     public const string PUNCTUATION = ":,=/><(-.+);*";
+    // List of letters
+    public const string LETTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+    // Dictionary of reserved words
+    public static readonly Dictionary<string, int> DICTIONARY = new Dictionary<string, int>() 
+    {
+        {"and", 1}, {"begin", 2}, {"Boolean", 3}, {"div", 4}, {"do", 5}, {"downto", 6},
+        {"else", 7}, {"end", 8}, {"false", 9}, {"fixed", 10}, {"float", 11}, {"for", 12},
+        {"function", 13}, {"if", 14}, {"integer", 15}, {"mod", 16}, {"not", 17}, 
+        {"or", 18}, {"procedure", 19}, {"program", 20}, {"read", 21}, {"repeat", 22},
+        {"string", 23}, {"then", 24}, {"true", 25}, {"to", 26}, {"type", 27}, {"until", 28}, 
+        {"var", 29}, {"while", 30}, {"write", 31}, {"writeln", 32}
```

Comment:
change these to use their TOKEN enumerators since we already have those set up; also DICTIONARY may not be a very descriptive name; I'd go with something like RESERVED_WORDS
'''
public static readonly Dictionary<string, int> RESERVED_WORDS = new Dictionary<string, int>() {
    {"and", TOKENS.AND}, // etc
}
'''